test/fixtures/krb5kdc-fixture/src/main/java/org/elasticsearch/test/fixtures/krb5kdc/Krb5kDcContainer.java

```diff
@@ -0,0 +1,172 @@
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public final class Krb5kDcContainer extends DockerEnvironmentAwareTestContainer {
+    public static final String DOCKER_BASE_IMAGE = "docker.elastic.co/elasticsearch-dev/krb5dc-fixture:1.0";
+    private final TemporaryFolder temporaryFolder = new TemporaryFolder();
+    private final ProvisioningId provisioningId;
+    private Path krb5ConfFile;
+    private Path keytabFile;
+    private Path esKeytabFile;
+
+    public enum ProvisioningId {
+        HDFS(
+            "hdfs",
+            "/fixture/src/main/resources/provision/hdfs.sh",
+            "/fixture/build/keytabs/hdfs_hdfs.build.elastic.co.keytab",
+            "/fixture/build/keytabs/elasticsearch.keytab",
+            "hdfs/hdfs.build.elastic.co@BUILD.ELASTIC.CO"
+        ),
+        PEPPA(
+            "peppa",
+            "/fixture/src/main/resources/provision/peppa.sh",
+            "/fixture/build/keytabs/peppa.keytab",
+            "/fixture/build/keytabs/HTTP_localhost.keytab",
+            "peppa@BUILD.ELASTIC.CO"
+        );
+
+        private final String id;
+        private final String scriptPath;
+        private final String keytabPath;
+        public final String esKeytab;
+        private final String keytabPrincipal;
+
+        ProvisioningId(String id, String scriptPath, String keytabPath, String esKeytab, String principal) {
+            this.id = id;
+            this.scriptPath = scriptPath;
+            this.keytabPath = keytabPath;
+            this.esKeytab = esKeytab;
+            this.keytabPrincipal = principal;
+        }
+    }
+
+    public Krb5kDcContainer() {
+        this(ProvisioningId.HDFS);
+    }
+
+    public Krb5kDcContainer(ProvisioningId provisioningId) {
+        super(new RemoteDockerImage(DOCKER_BASE_IMAGE));
+        this.provisioningId = provisioningId;
+        withNetwork(Network.newNetwork());
+        addExposedPorts(88, 4444);
+        withCreateContainerCmdModifier(cmd -> {
+            // Add previously exposed ports and UDP port
+            List<ExposedPort> exposedPorts = new ArrayList<>();
+            for (ExposedPort p : cmd.getExposedPorts()) {
+                exposedPorts.add(p);
+            }
+            exposedPorts.add(ExposedPort.udp(88));
+            cmd.withExposedPorts(exposedPorts);
+
+            // Add previous port bindings and UDP port binding
+            Ports ports = cmd.getPortBindings();
+            ports.bind(ExposedPort.udp(88), Ports.Binding.empty());
+            cmd.withPortBindings(ports);
+        });
+        withNetworkAliases("kerberos.build.elastic.co", "build.elastic.co");
+        withCopyFileToContainer(MountableFile.forHostPath("/dev/urandom"), "/dev/random");
+        withExtraHost("kerberos.build.elastic.co", "127.0.0.1");
+        withCommand("bash", provisioningId.scriptPath);
```

Comment:
ideally we would be able to get this executed as part of the image build but I haven't found a way as it requires certain hostname settings not available at build time